app/code/Magento/Customer/Model/Session.php

```diff
@@ -304,7 +309,11 @@ public function setCustomer(Customer $customerModel)
     public function getCustomer()
     {
         if ($this->_customerModel === null) {
-            $this->_customerModel = $this->_customerFactory->create()->load($this->getCustomerId());
+            $this->_customerModel = $this->_customerFactory->create();
+
+            if ($this->getCustomerId()) {
+                $this->_customerModel->load($this->getCustomerId());
```

Comment:
It would be also great to use `_customerResource` for loading the customer model if possible.

Reply:
> It would be also great to use `_customerResource` for loading the customer model if possible.

Looks like we can not use `_customerResource` and `customerRepository` because
`Magento\Customer\Model\Session`::`getCustomer()` should return instance of `\Magento\Customer\Model\Customer`.

So, `_customerResource` returns `\Magento\Customer\Model\ResourceModel\Customer` and `customerRepository` returns `\Magento\Customer\Api\Data\CustomerInterface` or `Magento\Customer\Model\Data\Customer`.

Thank you!

Reply:
> So, _customerResource returns \Magento\Customer\Model\ResourceModel\Customer

It's normal behavior. The resource model's load method is responsible for loading the model but you should not expect this model to be returned as a result.
The following example illustrates what I mean:
```
$this->_customerResource->load($this->_customerModel, $this->getCustomerId());
return $this->_customerModel;
```

Reply:
**implemented**

Thank you!